crates/orderbook/src/database/orders.rs

```diff
@@ -220,56 +220,95 @@ const ORDERS_FROM: &str = "\
     orders o \
 ";
 
-#[async_trait::async_trait]
-impl OrderStoring for Postgres {
-    async fn insert_order(&self, order: &Order) -> Result<(), InsertionError> {
-        const QUERY: &str = "\
+async fn insert_order(
+    order: &Order,
+    transaction: &mut sqlx::Transaction<'_, sqlx::Postgres>,
+) -> Result<(), InsertionError> {
+    const QUERY: &str = "\
             INSERT INTO orders (
                 uid, owner, creation_timestamp, sell_token, buy_token, receiver, sell_amount, buy_amount, \
                 valid_to, app_data, fee_amount, kind, partially_fillable, signature, signing_scheme, \
                 settlement_contract, sell_token_balance, buy_token_balance, full_fee_amount) \
             VALUES ($1, $2, $3, $4, $5, $6, $7, $8, $9, $10, $11, $12, $13, $14, $15, $16, $17, $18, $19);";
-        let receiver = order
-            .order_creation
-            .receiver
-            .map(|address| address.as_bytes().to_vec());
-        sqlx::query(QUERY)
-            .bind(order.order_meta_data.uid.0.as_ref())
-            .bind(order.order_meta_data.owner.as_bytes())
-            .bind(order.order_meta_data.creation_date)
-            .bind(order.order_creation.sell_token.as_bytes())
-            .bind(order.order_creation.buy_token.as_bytes())
-            .bind(receiver)
-            .bind(u256_to_big_decimal(&order.order_creation.sell_amount))
-            .bind(u256_to_big_decimal(&order.order_creation.buy_amount))
-            .bind(order.order_creation.valid_to)
-            .bind(&order.order_creation.app_data.0[..])
-            .bind(u256_to_big_decimal(&order.order_creation.fee_amount))
-            .bind(DbOrderKind::from(order.order_creation.kind))
-            .bind(order.order_creation.partially_fillable)
-            .bind(&*order.order_creation.signature.to_bytes())
-            .bind(DbSigningScheme::from(
-                order.order_creation.signature.scheme(),
-            ))
-            .bind(order.order_meta_data.settlement_contract.as_bytes())
-            .bind(DbSellTokenSource::from(
-                order.order_creation.sell_token_balance,
-            ))
-            .bind(DbBuyTokenDestination::from(
-                order.order_creation.buy_token_balance,
-            ))
-            .bind(u256_to_big_decimal(&order.order_meta_data.full_fee_amount))
-            .execute(&self.pool)
-            .await
-            .map(|_| ())
-            .map_err(|err| {
-                if let sqlx::Error::Database(db_err) = &err {
-                    if let Some(Cow::Borrowed("23505")) = db_err.code() {
-                        return InsertionError::DuplicatedRecord;
-                    }
+    let receiver = order
```

Comment:
Just to confirm, there are no changes besides indentation here right?

Reply:
That's right. I recommend ticking "ignore whitespace change" in the Github UI but sadly need to do it again for every PR.